.github/workflows/test-carthage-integration.yml

```diff
@@ -12,27 +12,28 @@ on:
 jobs:
 
   carthage:
-    runs-on: macos-12
+    runs-on: macos-latest
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - uses: n1hility/cancel-previous-runs@v3
       with:
         token: ${{ secrets.GITHUB_TOKEN }}
 
     - name: Select latest Xcode
       uses: maxim-lobanov/setup-xcode@v1
       with:
-        xcode-version: latest-stable
+        xcode-version: '16.1'
 
     - name: Resolve dependencies
       run: |
         brew update
         brew install xcodegen
+        brew install carthage
         brew upgrade carthage
```

Comment:
is upgrade carthage needed when you just installed it?

Reply:
It is a good practice. At least, from the literature online